apps/tools/tests/test_test_course.py

```diff
@@ -0,0 +1,55 @@
+import collections
+import json
+import os
+
+from librelingo_json_export.export import export_course
+from librelingo_yaml_loader.yaml_loader import load_course
+
+Settings = collections.namedtuple(
+    "Settings",
+    [
+        "dry_run",
+    ],
+)
+
+
+def test_course(tmpdir):
+    root = os.path.abspath(__file__)
+    root = os.path.dirname(root)
+    root = os.path.dirname(root)
+    root = os.path.dirname(root)
+    root = os.path.dirname(root)
+    print(root)
+
+    path_to_course = os.path.join(root, "courses", "test")
+    course = load_course(path_to_course)
+
+    settings = Settings(
+        dry_run=False,
+    )
+    output_path = str(tmpdir)
+    expected_path = os.path.join(root, "apps", "web", "src", "courses", "test")
+    print(output_path)
+
+    export_course(output_path, course, settings)
+    for dirname, _, files in os.walk(output_path):
+        for filename in files:
+            relpath = os.path.join(dirname[len(output_path) + 1 :], filename)
+
+            if filename.endswith(".json"):
+                with open(os.path.join(output_path, relpath)) as fh:
+                    actual = json.load(fh)
+                with open(os.path.join(expected_path, relpath)) as fh:
+                    expected = json.load(fh)
+                assert actual == expected, relpath
+                continue
+
+            if filename.endswith(".md"):
+                with open(os.path.join(output_path, relpath)) as fh:
+                    actual = fh.read()
+                with open(os.path.join(expected_path, relpath)) as fh:
+                    expected = fh.read()
+                assert actual == expected, relpath
+                continue
+
+            assert False, f"Unhandled file {relpath}"
```

Comment:
it could even be a simple exception, does not need to be an assertion. but it's good like this too

Reply:
Well, as I understand an assert is just an easy way to generate an exception on a condition.

Reply:
sure, I just wasn't sure if it will create the proper exception class in this case or not. I was working with the assumption that this is basically signalling an issue with the test itself, not with the unit under test. But I thought I bit more and I realized that it can also be because the functionality of that function was changed and generates new files.

so it's perfect like this :100: